modules/stochastic_tools/include/state_sim/model/events/StateSimEvFailRate.h

```diff
@@ -0,0 +1,45 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMEVFAILRATE_H
+#define STATESIMEVFAILRATE_H
+
+#include "StateSimEvTimeBased.h"
+#include "StateSimEvent.h"
+
+/**
+ * A Failure Rate event is a time based that samples the next time the event occures over a normal distribution using the given perameters
+ */
+class StateSimEvFailRate: public StateSimEvTimeBased
+{
+protected:
+  double _lambda;
+  TimespanH _time_rate;
+  TimespanH _comp_mission_time;
+
+public:
+  /**
+   * This constructor should only to be used by the auto model generation from the JSON file.
+   * used to create a temporary holder with the data to be filled later.
+  */
+  StateSimEvFailRate(std::string name);
```

Comment:
Be careful about method signatures like this. You should almost always prefer `const std::string & name` instead. Copying strings is expensive.